### PR TITLE
Lua API read-only access to the actual loadout

### PR DESCRIPTION
**Pull request by MjnMixael**
Provides read-only access to the actual loadout used to setup the mission. A later PR will add the ability to write to these values.

### CODE REVIEW

File: code/scripting/api/objs/shipwepselect.cpp [hunk 1]
```diff
@@ -119,6 +143,137 @@ ADE_VIRTVAR(ShipClassIndex,
...
+		return ADE_RETURN_NIL;
+
+	// Will eventually be writable so that Lua can manipulate the player loadout!
+	// Note may need to do some checking that the bank is valid for the current ship
+	// and that the amount-to-write is the correct size for the weapon/ship cargo size
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "This property is read only.");
+	}
+
+	if (idx < 1 || idx > 7) {
```
Comment (The-E) [inline code/scripting/api/objs/shipwepselect.cpp:169]:
```suggestion
	if (idx < 1 || idx > MAX_SHIP_WEAPONS) {
```
Leaving a magic number like that is just going to lead to issues when someone increases the number of primary or secondary banks

File: code/scripting/api/objs/shipwepselect.cpp [hunk 2]
```diff
@@ -119,6 +143,137 @@ ADE_VIRTVAR(ShipClassIndex,
...
+		return ADE_RETURN_NIL;
+	};
+	idx--; // Convert to Lua's 1 based index system
+
+	return ade_set_args(L, "i", current.getBank()->wep_count[idx]);
+}
+
+ADE_FUNC(__len, l_Loadout_Amount, nullptr, "The number of weapon banks in the slot", "number", "The number of banks.")
+{
+	return ade_set_args(L, "i", MAX_SHIP_PRIMARY_BANKS + MAX_SHIP_SECONDARY_BANKS);
```
Comment (The-E) [inline code/scripting/api/objs/shipwepselect.cpp:179]:
```suggestion
	return ade_set_args(L, "i", MAX_SHIP_WEAPONS);
```
MAX_SHIP_WEAPONS is defined as the sum of PRIMARY_BANKS and SECONDARY_BANKS

File: code/scripting/api/objs/shipwepselect.cpp [hunk 3]
```diff
@@ -119,6 +143,137 @@ ADE_VIRTVAR(ShipClassIndex,
...
+	if (!ade_get_args(L, "oi", l_Loadout_Weapon.Get(&current), &idx))
+		return ADE_RETURN_NIL;
+
+	// Will eventually be writable so that Lua can manipulate the player loadout!
+	// Note may need to do some checking that the bank is valid for the current ship
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "This property is read only.");
+	}
+
+	if (idx < 1 || idx > 7) {
```
Comment (The-E) [inline code/scripting/api/objs/shipwepselect.cpp:203]:
```suggestion
	if (idx < 1 || idx > MAX_SHIP_WEAPONS) {
```

File: code/scripting/api/objs/shipwepselect.cpp [hunk 4]
```diff
@@ -119,6 +143,137 @@ ADE_VIRTVAR(ShipClassIndex,
...
+		return ADE_RETURN_NIL;
+	};
+	idx--; // Convert to Lua's 1 based index system
+
+	return ade_set_args(L, "i", current.getBank()->wep[idx] + 1);
+}
+
+ADE_FUNC(__len, l_Loadout_Weapon, nullptr, "The number of weapon banks in the slot", "number", "The number of banks.")
+{
+	return ade_set_args(L, "i", MAX_SHIP_PRIMARY_BANKS + MAX_SHIP_SECONDARY_BANKS);
```
Comment (The-E) [inline code/scripting/api/objs/shipwepselect.cpp:213]:
```suggestion
	return ade_set_args(L, "i", MAX_SHIP_WEAPONS);
```

**Review by The-E — CHANGES_REQUESTED** [review]

File: code/scripting/api/objs/shipwepselect.cpp (hunk 1 above)
Comment (MjnMixael) [reply code/scripting/api/objs/shipwepselect.cpp]: Ah, I missed that that one existed or I would have used it. Done.

**Review by The-E — APPROVED** [review]

**Pull request merged**